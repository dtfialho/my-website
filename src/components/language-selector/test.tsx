import { screen, render } from '@testing-library/react'

import LanguageSelector from './'

const useRouter = jest.spyOn(require('next/router'), 'useRouter')
const defaultLocale = 'pt-BR'

describe('components/LanguageSelector', () => {
  it('Should match snapshot', () => {
    useRouter.mockImplementation(() => ({
      locale: defaultLocale
    }))

    const { container } = render(<LanguageSelector />)

    expect(container).toMatchSnapshot()
  })

  it('Should render the correct active language flag', () => {
    useRouter.mockImplementation(() => ({
      locale: defaultLocale
    }))

    render(<LanguageSelector />)

    expect(
      screen.getByAltText(`Active language ${defaultLocale}`)
    ).toBeInTheDocument()
  })
})
