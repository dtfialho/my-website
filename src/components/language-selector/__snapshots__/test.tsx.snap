// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/LanguageSelector Should match snapshot 1`] = `
.c0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: none;
  background: none;
  cursor: pointer;
}

<div>
  <button
    class="c0"
    type="button"
  >
    <img
      alt="Active language pt-BR"
      data-nimg="1"
      decoding="async"
      height="22"
      loading="lazy"
      src="/_next/image?url=%2Fimg%2Fpt-BR.jpg&w=64&q=75"
      srcset="/_next/image?url=%2Fimg%2Fpt-BR.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg%2Fpt-BR.jpg&w=64&q=75 2x"
      style="color: transparent;"
      width="29"
    />
  </button>
</div>
`;
